File: _정답코드/5주차/01_구간_합_구하기_2/solution.py
def solution(numbers, queries):
    n = len(numbers)
    prefixSum = [0] * (n + 1)
    
    for i in range(n):
        prefixSum[i + 1] = prefixSum[i] + numbers[i]
    
    result = []
    for i in range(len(queries)):
        l = queries[i][0]
        r = queries[i][1]
        result.append(prefixSum[r] - prefixSum[l - 1])
    
    return result

File: _정답코드/5주차/01_구간_합_구하기_2/Solution.java
class Solution {
    public int[] solution(int[] numbers, int[][] queries) {
        int n = numbers.length;
        int[] prefixSum = new int[n + 1];
        
        for (int i = 0; i < n; i++) {
            prefixSum[i + 1] = prefixSum[i] + numbers[i];
        }
        
        int[] result = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int l = queries[i][0];
            int r = queries[i][1];
            result[i] = prefixSum[r] - prefixSum[l - 1];
        }
        
        return result;
    }
}

